transport/src/main/java/io/netty/channel/SingleThreadEventLoop.java

```diff
@@ -154,4 +157,61 @@ public int pendingTasks() {
     public int registeredChannels() {
         return -1;
     }
+
+    /**
+     * @return read-only iterator of active {@link Channel}s registered with this {@link EventLoop}.
+     *         The returned value is not guaranteed to be exact accurate and
+     *         should be viewed as a best effort.
+     * @throws UnsupportedOperationException if operation is not supported by implementation.
+     */
+    @UnstableApi
+    public Iterator<Channel> registeredChannelsIterator() {
```

Comment:
Add to the docs that this is only expected to be called from within  the event loop.